media-gfx/opencsg/opencsg-1.5.1.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit qmake-utils
+
+MY_P="OpenCSG-${PV}"
+
+DESCRIPTION="The Constructive Solid Geometry rendering library"
+HOMEPAGE="https://www.opencsg.org"
+SRC_URI="https://www.opencsg.org/${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="GPL-2+"
+SLOT="0"
```

Comment:
This is still 0...?

Reply:
Is there a reason why it should change?

Reply:
ABI changed, so the subslot needs to represent that to trigger rebuilds? Or am I misunderstanding what happened here(

Reply:
Isn't this why I added the subslot operator? Should the update in version number together with this operator enough to trigger rebuilds?

Reply:
Subslot operators do nothing if the subslot didn't change. They don't work based on version.

Reply:
Eeh woops, indeed. I'll sort the tree!

Reply:
Thanks!

Reply:
So yeah @waebbl `SLOT="0/1.5" 
0 = slot operator,
1.5 = subslot

when subslot changes it means the new SONAME is incompatible and every ebuild depending on your `package:=` will get rebuilt. Think it this way: openscad is now **bound** to opencsg:0/1.5. 

Whenever you update opencsg you should check if the SONAME has changed, and/or if the new libs break the API/ABI. 
```
# objdump -p /usr/lib64/libopencsg.so.1.5.1 | grep SONAME
  SONAME               libopencsg.so.1
```
note that it's still called "1" but I named it "1.5" in anticipation of it breaking with 1.6 release. If 1.6.0 is fine, you can **keep** the "0/1.5" slot. That is perfectly fine. 

Here's how you easiest check the SONAME API/ABI breakage:

> emerge -1av iwdevtools libabigail
> (if you're not, *cd* to your ebuild development directory ../media-gfx/opencsg)
> ebuild opencsg-oldversion clean install
> ebuild opencsg-newversion clean install
> qa-cmp -I media-gfx/opencsg
```
 CMP: media-gfx/opencsg-1.4.2-r1/image with media-gfx/opencsg-1.5.1/image
 FILES:-usr/lib64/libopencsg.so.1.4
 FILES:+usr/lib64/libopencsg.so.1.5
   ABI: libopencsg.so.1(64) func(-1) [BREAKING]
------> FILES(+1,-1) ABI(-1,>B<)
```

https://devmanual.gentoo.org/general-concepts/slotting/index.html#sub-slots

Reply:
Thanks @thesamesam and @juippis for fixing this on the fly and detailed explanation.